examples/README.md

```diff
@@ -40,35 +39,31 @@ Evaluate expression `i.greet(you)` with:
 ```
 
 First we need to declare two string variables and `greet` function.
-`NewInstanceOverload` must be used if we want to declare function which will
-operate on a type. First element of slice passed as `argTypes` into
-`NewInstanceOverload` is declaration of instance type. Next elements are
-parameters of function.
+`Function` must be used if we want to declare an extension to CEL. The
+`MemberOverload` declares an overload id, a list of arguments where the
+first element the `argTypes` slice is the target type of the member
+function. The remaining argument types are the signature of the member
+method.
 
 ```go
-    decls.NewVar("i", decls.String),
-    decls.NewVar("you", decls.String),
-    decls.NewFunction("greet",
-        decls.NewInstanceOverload("string_greet_string",
-            []*exprpb.Type{decls.String, decls.String},
-            decls.String))
-    ... // Create env and compile
-```
-
-Let's implement `greet` function and pass it to `program`. We will be using
-`Binary`, because `greet` function uses 2 parameters (1st instance, 2nd
-function parameter).
-
-```go
-    greetFunc := &functions.Overload{
-        Operator: "string_greet_string",
-        Binary: func(lhs ref.Val, rhs ref.Val) ref.Val {
-            return types.String(
-                fmt.Sprintf("Hello %s! Nice to meet you, I'm %s.\n", rhs, lhs))
-            }}
-    prg, err := env.Program(c, cel.Functions(greetFunc))
-
-    out, _, err := prg.Eval(map[string]interface{}{
+    env, _ := cel.NewEnv(
+        cel.Variable("i", cel.StringType),
+        cel.Variable("you", cel.StringType),
+        cel.Function("greet",
+            cel.MemberOverload("string_greet_string",
+                []*cel.Type{cel.StringType, cel.StringType},
+                cel.StringType,
+                cel.BinaryBinding(func (lhs, rhs ref.Val) ref.Val {
+                    return types.String(
+                        fmt.Sprintf("Hello %s! Nice to meet you, I'm %s.\n", rhs, lhs))
+                    },
+                ),
+            ),
+        ),
+    )
+    // Create env and compile
```

Comment:
Nit: Looks like we can just remove this comment or update

Reply:
Done.